### PR TITLE
 Create SWC-137: Hardcoded gas limits

### PR DESCRIPTION
**Pull request by b-mueller**
[See also](https://diligence.consensys.net/blog/2019/09/stop-using-soliditys-transfer-now/)

### CODE REVIEW

File: entries/SWC-137.md [hunk 1]
```diff
@@ -0,0 +1,20 @@
+# Title
+Message call with hardcoded gas amount
+
+## Relationships
+[CWE-655: Improper Initialization](https://cwe.mitre.org/data/definitions/665.html)
```
Comment (thec00n) [inline entries/SWC-137.md]: Is this more accurate https://cwe.mitre.org/data/definitions/664.html?

File: entries/SWC-137.md [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+
+## Description
+
+The `transfer()` and `send()` functions forward a fixed amount of 2300 gas. Historically, it has often been recommended to use these functions for value transfers to guard against reentrancy attacks. However, the gas cost of EVM instructions may change significantly during hard forks which may break already deployed contract systems that make fixed assumptions about gas costs.  For example. [EIP 1884](https://eips.ethereum.org/EIPS/eip-1884) broke several existing smart contracts due to a cost increase of the SLOAD instruction. 
+
+## Remediation
+
+Avoid the use of `transfer()` and `send()` and do not otherwise specify a fixed amount of gas when performing calls. Use `.call.value(...)("")` instead. Use the checks-effects-interactions pattern to prevent reentrancy attacks.
```
Comment (thec00n) [inline entries/SWC-137.md]: To be consistent with https://github.com/SmartContractSecurity/SWC-registry/pull/240 should reentrancy locks be mentioned here?

File: entries/SWC-137.md (hunk 1 above)
Comment (b-mueller) [reply entries/SWC-137.md]: I didn't find anything that really fit... by "resource" they're referring to Java class instances and things like that. Do you think that's more appropriate than using the broad category?

**Comment by maurelian** [comment]
Will this only detect the case where a gas limit is naively set as a constant value? 

It's common in metatransaction's situations to have a gas limit set by argument ([example](https://github.com/arcadeum/multi-token-standard/blob/master/contracts/tokens/ERC1155/ERC1155Meta.sol#L176)).

**Comment by b-mueller** [comment]
@maurelian yes, we only flag gas limit set as a constant value (see code samples).

**Pull request merged**